diff --git a/auto/cc/conf b/auto/cc/conf
index bdcce2c..effa360 100644
--- a/auto/cc/conf
+++ b/auto/cc/conf
@@ -2,7 +2,7 @@
 # Copyright (C) Igor Sysoev
 
 
-LINK="\$(CC)"
+LINK="\$(CC) -rdynamic"
 
 ngx_include_opt="-I "
 ngx_compile_opt="-c"
diff --git a/auto/modules b/auto/modules
index a8607ee..58e3d05 100644
--- a/auto/modules
+++ b/auto/modules
@@ -471,11 +471,24 @@ done
 echo                                          >> $NGX_MODULES_C
 echo 'ngx_module_t *ngx_modules[] = {'        >> $NGX_MODULES_C
 
+have_so_module="NO"
 for mod in $modules
 do
     echo "    &$mod,"                         >> $NGX_MODULES_C
+    if [ $mod = ngx_http_so_module ]; then
+        have_so_module="YES"
+    fi
 done
 
+so_module_slot=16 # by default
+i=$so_module_slot
+if [ $have_so_module = YES ]; then
+    while [ ${i} -gt 0 ]; do
+        echo "    NULL,"                      >> $NGX_MODULES_C
+        let i=${i}-1;
+    done
+fi
+
 cat << END                                    >> $NGX_MODULES_C
     NULL
 };
